### DISCUSSION TITLE
Aspire Integration Tests are failing ONLY when run via devops agent.

### DISCUSSION THREAD

**Discussion by wolkowsky** [microsoft/aspire#5051] [Q&A] [answered]
Hi, 

We've been fighting with weird looking issue, we do have an simple test: 

```
[Theory]
[MemberData(nameof(AppHostAssemblies))]
public async Task CustomerApiHealthReturnsOk(string appHostPath)
{
    // Arrange
    var appHost = await DistributedApplicationTestFactory.CreateAsync(appHostPath, output);

    await using var app = await appHost.BuildAsync();

    var appHostLogs = app.GetAppHostLogs();
    var resourceLogs = app.GetResourceLogs();

    await app.StartAsync(waitForResourcesToStart: true);

    appHostLogs.EnsureNoErrors();
    resourceLogs.EnsureNoErrors(ShouldAssertErrorsForResource);

    // Act
    var httpClient = app.CreateHttpClient(Common.ServiceConstants.CustomerApiServiceName);
    var response = await httpClient.GetAsync("/health");

    // Assert
    Assert.Equal(HttpStatusCode.OK, response.StatusCode);

    await app.StopAsync();
}
```

It all works perfectly fine when run locally via VS or terminal. It also works fine (when we ssh into this) on VM's and Virtual scale sets that we do use as our build agent, however, it does not work when the tests is being run via agent and azdo pipelines. 

All I can see in the logs is: 

 `2024-07-23T13:22:35.6274924+00:00 fail [Aspire.Hosting.Dcp.dcpctrl.ExecutableReconciler] the Executable cannot be run and will be marked as finished	{"Executable": {"name":"operatorapi_6fd2ce7c-dgbotvo"}, "Reconciliation": 14, "error": "no runner found for execution type 'IDE'"}`
But I don't really know what's meant by that and what I need to change in terms of my build agent or pipeline.
Have anyone seen it before? What are we missing? 

EDIT: 
Node apps are starting fine. .NET api's are not. 
EDIT2: It's Linux (Ubuntu 22) self hosted agent.

**Comment by wolkowsky** [comment, marked as answer]
Looks like something in build agent sets `DEBUG_SESSION_PORT` it Results in DCP for the App Host trying to use a none existent IDE.

Setting this to '' helps - dunno yet if there are any consequences of erasing that value but the tests are working fine. 
Thank you @Jamie-round455 for finding the solution!

**Reply by davidfowl** [reply]
Do you know what is setting this?

**Reply by Jamie-round455** [reply]
We are reasonably certain it's the DevOps agent that sets it. 
Running the same process in: Windows; WSL (Ubuntu 22.04); Dev container; Ubuntu VM; environments work as expected. 
Using both the Microsoft-hosted agents and our own custom agent ran into the same error. When we ran tests that simply echo the value of that variable, without any other steps, it was always set.

**Reply by davidfowl** [reply]
cc @karolz-ms @dbreshears

**Reply by karolz-ms** [reply]
I suspect this is some sort of configuration copy-paste mistake. The name of the variable and its function are too specific to Aspire for the DevOps agent to just "get an idea to start setting it" on its own.

**Reply by wolkowsky** [reply]
> I suspect this is some sort of configuration copy-paste mistake. 

Yea, you were right. It turned out that someone have set this variable in the pipeline. Anyway - thank you guys. Problem solved.
